include/bluetooth/mesh/access.h

```diff
@@ -658,13 +659,14 @@ static inline bool bt_mesh_model_in_primary(const struct bt_mesh_model *mod)
  *
  *  @param mod      Mesh model.
  *  @param vnd      This is a vendor model.
+ *  @param key      Unique identification of model data
  *  @param data     Model data to store, or NULL to delete any model data.
  *  @param data_len Length of the model data.
  *
  *  @return 0 on success, or (negative) error code on failure.
  */
 int bt_mesh_model_data_store(struct bt_mesh_model *mod, bool vnd,
-			     const void *data, size_t data_len);
+			     u16_t key, const void *data, size_t data_len);
```

Comment:
This key should be a string as well, to stay consistent with the rest of the settings API, in my opinion. The max length of a single settings entry name is 7 characters, so if this function only accepts a single level of data, it shouldn't introduce any complexity. 

This also opens up for passing `key=NULL` to get the same single-entry functionality as the current API.